x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/lib/utils.test.ts

```diff
@@ -62,4 +62,77 @@ describe('utils', () => {
       expect(isStateValid(components)).toBe(false);
     });
   });
+
+  describe('extractMappingsDefinition', () => {
+    test('should detect that the mappings has multiple types and return null', () => {
+      const mappings = {
+        type1: {
+          properties: {
+            name1: {
+              type: 'keyword',
+            },
+          },
+        },
+        type2: {
+          properties: {
+            name2: {
+              type: 'keyword',
+            },
+          },
+        },
+      };
+
+      expect(extractMappingsDefinition(mappings)).toBe(null);
+    });
+
+    test('should detect that the mappings has one type under the "_doc" type', () => {
```

Comment:
I can add it but I don't think it is necessary because the logic does not care about the name of the type. It only looks if there is more than 1 (of any name).

Reply:
Actually I made me see that there is still a case where it could break. When one of the types has not defined any properties. It might be a super edge case, but this test currently fails:

```js
test('should detect that the mappings has multiple types even when one of the type has not defined any properties', () => {
  const mappings = {
    type1: {
      source: {
        excludes: [],
        includes: [],
        enabled: true,
      },
      _routing: {
        required: false,
      },
    },
    type2: {
      properties: {
        name2: {
          type: 'keyword',
        },
      },
    },
  };

  expect(extractMappingsDefinition(mappings)).toBe(null);
});
```